crates/paas/src/worker.rs

```diff
@@ -0,0 +1,202 @@
+//! Worker pool implementation
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+
+use tokio::{spawn, time::sleep};
+use tracing::{debug, error, info};
+
+use crate::{error::ProverServiceError, state::ProverServiceState, task::TaskStatus, Prover};
+
+/// Worker pool that processes tasks for all backends
+pub struct WorkerPool<P: Prover> {
+    state: Arc<ProverServiceState<P>>,
+}
+
+impl<P: Prover> WorkerPool<P> {
+    /// Create a new worker pool
+    pub fn new(state: Arc<ProverServiceState<P>>) -> Self {
+        Self { state }
+    }
+
+    /// Main worker pool loop
+    pub async fn run(self) {
```

Comment:
Don't like having `run` functions like this as it kinda conflates the "worker state" and supplementary things like input handles.  Not sure if that applies here.

Can this use the service framework?  Or could it use a more general threadpool?